src/db-api.js

```diff
@@ -3,6 +3,69 @@
 import type { Pool, ResultSet } from 'pg'
 import type { DbApi } from 'icarus-backend'; // eslint-disable-line
 
+/**
+ * Checks if addresses array contains group address. According to Emurgo,
+ * frontend should not explicitly query group addresses, it can lead to hard
+ * to detect bugs.
+ * @param {Db Object} db
+ * @param {Array<Address>} addresses
+ */
+const hasGroupAddress = (db: Pool) => async (
```

Comment:
we don't want to check it this way. Instead, we should really check the addresses. I believe we can use more-or-less something like
```
isGroupAddress(addr) => isHex(addr) && (addr.startswith('04') || addr.startswith('84'))
```